### ISSUE TITLE
Command substitution - more weirdness

### ISSUE THREAD

**Issue by cunneen** [dotenvx/dotenvx#454]
Quite likely related to #453 .

### My `.env` file:
  ```env
  FOO=$(echo 'This should have a value of ﹩PWD because it is in single-quotes: $PWD')
  BAR=$(echo This should say hello there: hello $(echo there))
  BUZZ=$(jq -c . eslintrc.json)
  ```

  > NOTE: The first dollar-sign in the value of `FOO` above is not actually a dollar-sign, but rather an emoji, in order to ensure it doesn't confuse the substitution process.

`eslintrc.json`:
  ```json
  {
    "$schema": "https://json.schemastore.org/eslintrc.json",
    "rules": {"@typescript-eslint/no-explicit-any":"error"}
  }  
  ```

### Expected result:
I expected `dotenvx` to perform command substitution similar to how a shell would do it i.e.:
  ```sh
  bash-5.2$ source .env
  bash-5.2$ for i in FOO BAR BUZZ; do echo "${i}='${!i}'"; done
  ```

Expected Output:
  ```sh
  FOO='This should have a value of ﹩PWD because it is in single-quotes: $PWD'
  BAR='This should say hello there: hello there'
  BUZZ='{"$schema":"https://json.schemastore.org/eslintrc.json","rules":{"@typescript-eslint/no-explicit-any":"error"}}'
  ```

### Actual result

All three of the test cases had different problems:
  ```sh
  bash-5.2$ dotenvx get -pp
  ```

Actual output:
  ```sh
  {
    "FOO": "This should have a value of ﹩PWD because it is in single-quotes: /Users/mikecunneen/Development/dotenvx-bug",
    "BAR": "$(echo This should say hello there: hello there)",
    "BUZZ": "{\"\":\"https://json.schemastore.org/eslintrc.json\",\"rules\":{\"@typescript-eslint/no-explicit-any\":\"error\"}}"
  }
```

The most unexpected of these (for me) was the disappearance of the `$status` key in the `BUZZ` value!

To further illustrate (with the same input files): 
  ```sh
  bash-5.2$ schema="THE_ENV_VALUE_OF_SCHEMA" dotenvx get -pp
  ```
  
Output:
  ```json
  {
    "FOO": "This should have a value of ﹩PWD because it is in single-quotes: /Users/mikecunneen/Development/dotenvx-bug",
    "BAR": "$(echo This should say hello there: hello there)",
    "BUZZ": "{\"THE_ENV_VALUE_OF_SCHEMA\":\"https://json.schemastore.org/eslintrc.json\",\"rules\":{\"@typescript-eslint/no-explicit-any\":\"error\"}}"
  }
  ```

**Comment by motdotla** [comment]
related from #453 

---
What follows is just some more information about what I was trying to do. You can safely stop reading at this point.

### Background

I wanted to include a JSON document into a value in my `.env` file. But it has a `$schema` property that I don't want:

eslintrc.json
  ```json
  {
    "$schema": "https://json.schemastore.org/eslintrc.json",
    "rules": {"@typescript-eslint/no-explicit-any":"error"}
  }
  ```

I can remove the `$schema` property using `jq` :
  ```sh
  bash-5.2$ jq -c 'del(.["$schema"])' eslintrc.json 
  ```

Output: 
  ```json
  {"rules":{"@typescript-eslint/no-explicit-any":"error"}}
  ```

But trying to substitute this command in my `.env` file doesn't do what one would expect:

`.env`:
  ```env
  FOO = $(jq -c 'del(.["$schema"])' eslintrc.json)
  ```

Command:
  ```sh
  bash-5.2$ dotenvx get -pp
  ```

Actual Output:
  ```json
  {
    "FOO": "$(jq -c 'del(.[\"\"])' eslintrc.json)"
  }
  ```

Expected Output:
  ```json
  {
    "FOO": "{\"rules\":{\"@typescript-eslint/no-explicit-any\":\"error\"}}"
  }
  ```

You can see in the above "Actual Output" that there is another (presumably related) issue that I'll raise separately, whereby `dotenvx` attempts to substitute a value for `$status` even though in the input `.env` file, the expression is enclosed in single-quotes. 

To further illustrate:

Command:
  ```sh
  bash-5.2$ schema="THE_ENV_VALUE_OF_SCHEMA" dotenvx get -pp 
  ```

Actual Output:
  ```json
  {
    "FOO": "$(jq -c 'del(.[\"THE_ENV_VALUE_OF_SCHEMA\"])' eslintrc.json)"
  }
  ```

No amount of quote-fu or backslashery seem able to suppress this behaviour without breaking the syntax required for the `jq` command.

**Comment by motdotla** [comment]
this is fixed in `v1.24.3`:

```
BAR=$(echo This should say hello there: hello $(echo there))
```

---

the command substitution is actually working correctly for this:

```
FOO=$(echo 'This should have a value of ﹩PWD because it is in single-quotes: $PWD')
```

resulting in `This should have a value of ﹩PWD because it is in single-quotes: $PWD` BUT then the `variable expansion` runs immediately afterward expanding $PWD to empty string.

---

this is working currently as intended because `variable expansion` attempts to expand the $schema key.

```
BUZZ=$(jq -c . eslintrc.json)
```

**Comment by motdotla** [comment]
v1.24.4 should fix all this for you. let me know if not!

https://github.com/dotenvx/dotenvx/blob/main/CHANGELOG.md#1244
